project1/application.py

```diff
@@ -1,40 +1,90 @@
 import os
-
-from flask import Flask, session
+from flask import Flask, session,redirect,url_for
 from flask_session import Session
 from sqlalchemy import create_engine
 from sqlalchemy.orm import scoped_session, sessionmaker
 from flask import render_template, request
+from Users import *
+from datetime import datetime
+
 
 app = Flask(__name__)
 
-# Check for environment variable
-if not os.getenv("DATABASE_URL"):
-    raise RuntimeError("DATABASE_URL is not set")
+# # Check for environment variable
+# if not os.getenv("DATABASE_URL"):
+#     raise RuntimeError("DATABASE_URL is not set")
 
 # Configure session to use filesystem
-app.config["SESSION_PERMANENT"] = False
-app.config["SESSION_TYPE"] = "filesystem"
-Session(app)
+# app.config["SESSION_PERMANENT"] = False
+# app.config["SESSION_TYPE"] = "filesystem"
+# Session(app)
+app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URL")
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+db.init_app(app)
+app.secret_key = "session"
 
-# Set up database
-engine = create_engine(os.getenv("DATABASE_URL"))
-db = scoped_session(sessionmaker(bind=engine))
+# # Set up database
+# engine = create_engine(os.getenv("DATABASE_URL"))
+# db = scoped_session(sessionmaker(bind=engine))
 
 
 @app.route("/")
 def index():
-    return "Project 1: TODO"
+    # return "Project 1: TODO"
+    if session['email'] != None:
+    # if 'email' in session:
+        # email = session['email']
+        return render_template("login.html")
+    else:
+        return render_template("Register.html")
 
 @app.route("/register",methods=["GET","POST"])
 def register():
     if(request.method == "POST"):
+        Users.query.all()
         name = request.form.get("name")
         pwd = request.form.get("pwd")
         email = request.form.get("email")
-        print("name : " ,name)
-        print("password : ",pwd)
-        print("email: ",email)
-        return render_template("hello.html", name=name)
+        # gender = request.form.get("gender")
+        user_details = Users(name= name, email= email,password= pwd,datetime = datetime.now())
+        try:
+            db.session.add(user_details)
+            db.session.commit()
+            print("name : " ,name)
+            print("password : ",pwd)
+            print("email: ",email)
+            # print("gender: ",gender)
+            return render_template("hello.html", name=name)
+        except Exception:
+            return render_template("errorpage.html")
```

Comment:
Can you explain in which case you will get errorpage.html.

Reply:
I have used the email as primary key. So whenever the user tried to register with the email that was already registered then it will show the error message that Registration failed Please register again.  So in this case you will get errorpage.html

Reply:
It is good to send an error message to the user.

Reply:
Rest all looks fine.

Reply:
Did you update this error message part?

Reply:
Yes mam, I have updated

Reply:
did you commit the changes?check pls

Reply:
Yes mam, I have committed the changes Please check it